cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/SystemAppModule.java

```diff
@@ -68,6 +60,12 @@ protected void configure() {
     bind(ArtifactRepository.class).to(RemoteArtifactRepository.class);
     bind(PreferencesFetcher.class).to(RemotePreferencesFetcherInternal.class).in(Scopes.SINGLETON);
     bind(PluginFinder.class).to(RemoteWorkerPluginFinder.class);
+
+    bind(ArtifactLocalizerClient.class).in(Scopes.SINGLETON);
```

Comment:
Yes. The optional binding doesn’t bind the actual instance. It works before because we triggered a JIT binding before the optional binding was requested, but it is not always the case .